spirv_hlsl.cpp

```diff
@@ -6355,6 +6361,19 @@ void CompilerHLSL::emit_instruction(const Instruction &instruction)
 		statement("SetMeshOutputCounts(", to_unpacked_expression(ops[0]), ", ", to_unpacked_expression(ops[1]), ");");
 		break;
 	}
+	case OpEmitMeshTasksEXT:
+	{
+		if (instruction.length == 4)
+		{
+			statement("DispatchMesh(", to_unpacked_expression(ops[0]), ", ", to_unpacked_expression(ops[1]), ", ",
+			    to_unpacked_expression(ops[2]), ", ", to_unpacked_expression(ops[3]), ");");
+		}
+		else
+		{
+			SPIRV_CROSS_THROW("Amplification shader in HLSL must have payload");
```

Comment:
I've tested with DXC: empty struct as payload compiles, yet produces invalid byte-code
`bcf5408759f3.hlsl:17:5: error: For amplification shader with entry 'main', payload size 4 is greater than declared size of 0 bytes`
https://shader-playground.timjones.io/8d8dbf3b63aa67aa3d000327b56915dd

Reply:
Could you just make a dummy struct with one int?

Reply:
Haven't tested yet, but probably 'one-int' wont work, because task-payload should match mesh-payload in order to link.